server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -1290,36 +1281,92 @@ private void ensureMutable() {
      * The iterator then resumes on the first node by returning the second shard and continues until all shards from
      * all the nodes have been returned.
      */
-    public Iterator<ShardRouting> nodeInterleavedShardIterator() {
+    public Iterator<ShardRouting> nodeInterleavedShardIterator(boolean movePrimaryFirst) {
```

Comment:
Will be great to add more tests for this change

Reply:
I have added unit test for testing primary first shard iteration